bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -55,9 +60,53 @@ public static String getLocalIpv4HostAddress() {
             }
             return hostAddress;
         } catch (SocketException ex) {
-            LOGGER.error("Could not retrieve network interface: " + ex.getMessage(), ex);
+            LOGGER.error("Could not retrieve network interface: {}", ex.getMessage(), ex);
             return null;
         }
     }
 
+    /**
+     * Get all broadcast addresses on the current host
+     *
+     * @return list of broadcast addresses
```

Comment:
If the `SocketException` is thrown you return `null`.
So, this should be stated in the JavaDoc.
But wouldn't it make sense to throw the exception, so the caller is notified about the error -- or if the caller cannot handle it at all (does he need to know about the error) return an empty list?